### PR TITLE
Move Dart SDK to 1.15-dev.4 to support _embedder.yaml

### PR DESCRIPTION
**Pull request by johnmccutchan**

### CODE REVIEW

**Comment by johnmccutchan** [comment]
This pairs with https://github.com/flutter/engine/pull/2433

@Hixie @abarth

**Comment by abarth** [comment]
Spoke with @johnmccutchan .  I think we want to hold off on this patch until Wed.

File: bin/cache/update_dart_sdk.sh [hunk 1]
```diff
@@ -10,6 +10,12 @@ FLUTTER_ROOT=$(dirname $(dirname $(dirname "${BASH_SOURCE[0]}")))
 DART_SDK_PATH="$FLUTTER_ROOT/bin/cache/dart-sdk"
 DART_SDK_STAMP_PATH="$FLUTTER_ROOT/bin/cache/dart-sdk.stamp"
 DART_SDK_VERSION=`cat "$FLUTTER_ROOT/bin/cache/dart-sdk.version"`
+DART_CHANNEL="stable"
+
+if [[ $DART_SDK_VERSION == *"-dev."* ]]
+then
+  DART_CHANNEL="dev"
+fi
```
Comment (abarth) [inline bin/cache/update_dart_sdk.sh]: We can do this work inside in the "if" below so that we only compute the channel when the version needs updating.

**Comment by abarth** [comment]
LGTM, but let's hold off until Wed to land.

**Comment by abarth** [comment]
We're ready for this patch now.  Thanks!

**Pull request merged**